### PR TITLE
Remove Always approve workflow

### PR DESCRIPTION
**Pull request by bzwei**
https://projects.engineering.redhat.com/browse/SSP-856. 
If no workflow is linked to any resource in an order, the default is already always approve. There is no need to ask a user to specifically choose an always approve workflow.

### CODE REVIEW

File: db/migrate/20200226115345_delete_always_approve_from_workflows.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class DeleteAlwaysApproveFromWorkflows < ActiveRecord::Migration[5.2]
+  def up
+    always_approve = Workflow.find_by(:name => 'Always approve')
+    return unless always_approve
+
+    Request.where(:workflow_id => always_approve.id).update_all(:workflow_id => nil)
```
Comment (gmcculloug) [inline db/migrate/20200226115345_delete_always_approve_from_workflows.rb]: You can let Rails do some of the work for you here but referencing the object and not the `id`.

```ruby
Request.where(:workflow => always_approve).update_all(:workflow => nil)
```

File: spec/controllers/v1.0/graphql_controller_spec.rb [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
   let!(:template) { create(:template) }
   let(:template_id) { template.id }
-  let!(:workflows) { create_list(:workflow, 4, :template_id => template.id) }
+  let!(:workflows) { create_list(:workflow, 4, :template_id => template.id, :tenant_id => tenant.id) }
```
Comment (gmcculloug) [inline spec/controllers/v1.0/graphql_controller_spec.rb]: Same with these specs, you just need to reference the objects:
```ruby
let!(:workflows) { create_list(:workflow, 4, :template => template, :tenant => tenant) }
```

File: spec/controllers/v1.0/requests_controller_spec.rb [hunk 3]
```diff
@@ -8,7 +8,7 @@
   let(:headers_with_approver)  { default_headers.merge(Insights::API::Common::Request::PERSONA_KEY => described_class::PERSONA_APPROVER) }
   let(:headers_with_requester) { default_headers.merge(Insights::API::Common::Request::PERSONA_KEY => described_class::PERSONA_REQUESTER) }
 
-  let(:workflow) { create(:workflow, :name => 'Test always approve') }
+  let(:workflow) { create(:workflow, :name => 'Test always approve', :tenant_id => tenant.id) }
```
Comment (gmcculloug) [inline spec/controllers/v1.0/requests_controller_spec.rb]: same

File: spec/controllers/v1.0/workflows_controller_spec.rb [hunk 4]
```diff
@@ -5,7 +5,7 @@
 
   let!(:template) { create(:template) }
   let(:template_id) { template.id }
-  let!(:workflows) { create_list(:workflow, 16, :template_id => template.id) }
+  let!(:workflows) { create_list(:workflow, 16, :template_id => template.id, :tenant_id => tenant.id) }
```
Comment (gmcculloug) [inline spec/controllers/v1.0/workflows_controller_spec.rb]: same

File: db/migrate/20200226115345_delete_always_approve_from_workflows.rb (hunk 1 above)
Comment (bzwei) [reply db/migrate/20200226115345_delete_always_approve_from_workflows.rb]: Seems `update_all` does not follow relationships

Comment (gmcculloug) [reply db/migrate/20200226115345_delete_always_approve_from_workflows.rb]: Ok, but we can still update the `where` statement.

**Pull request merged**